src/main/scala/strawman/collection/Iterator.scala

```diff
@@ -127,6 +159,34 @@ trait Iterator[+A] extends IterableOnce[A] { self =>
     this
   }
 
+  /*
+   * Implemented by means of a buffer to keep the last n elements from being returned during iteration.
+   */
+  def dropRight(n: Int): Iterator[A] = {
```

Comment:
Unless I’m wrong in 2.12 we do have `takeRight` on `Iterable`: https://github.com/scala/scala/blob/v2.12.2/src/library/scala/collection/IterableLike.scala#L209-L227
(And it performs two traversals)